File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public class Square implements Shape {
    // add encapsulation
    private double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }

    // override the getArea method
    @Override
    public double getArea() {
        return sideLen * sideLen;
    }
}

File: typescript/src/shapes/square.ts
class Square implements Shape {
    sideLen: number;

    constructor(sideLen: number) {
        this.sideLen = sideLen;
    }

    // override the computeArea method
    computeArea(): number {
        return this.sideLen * this.sideLen
    }
}

export { Square }